module/VuFindSearch/src/VuFindSearch/Backend/Primo/Connector.php

```diff
@@ -603,44 +603,40 @@ protected function process($data, $params = [])
     /**
      * Retrieves a document specified by the ID.
      *
-     * @param string $recordId  The document to retrieve from the Primo API
-     * @param string $inst_code Institution code (optional)
-     * @param bool   $onCampus  Whether the user is on campus
+     * @param string  $recordId  The document to retrieve from the Primo API
+     * @param ?string $inst_code Institution code (optional)
+     * @param bool    $onCampus  Whether the user is on campus
      *
      * @throws \Exception
      * @return string    The requested resource
      */
-    public function getRecord($recordId, $inst_code = null, $onCampus = false)
+    public function getRecord(string $recordId, $inst_code = null, $onCampus = false)
     {
         // Query String Parameters
-        if (isset($recordId)) {
```

Comment:
I'm curious if @EreMaijala has any thoughts/concerns regarding this change. Would it be better to change it to a `strlen` check so we don't call the API if we're passed an empty string, or is there really no reason to have that empty response case in the code?

Reply:
Looks to me like putting this at the top of the function:

<pre>
if ('' !== $recordId) {
    return self::$emptyQueryResponse;
}
</pre>

would cover the case while making things more readable -- better to have an early return than huge amounts of logic inside an if statement. @xmorave2, what do you think?

Reply:
I do agree it is more readable and cleaner solution. I added it.

Reply:
Thanks!